apps/schedules/lib/hours_of_operation.ex

```diff
@@ -196,31 +220,94 @@ defmodule Schedules.HoursOfOperation do
   defp time(%{"departure_time" => nil, "arrival_time" => time}), do: time
   defp time(%{"departure_time" => time}), do: time
 
-  defp departure(%JsonApi{data: data}) do
-    {:ok, departure(data)}
+  # Grabs the head sign data from the stops,
+  # Letting us know the terminal stops for the trip
+  defp get_terminal_stops(data) do
+    data
+    |> Enum.map(fn data ->
+      trips = Map.get(data.relationships, "trip")
+      trip = List.first(trips)
+      Map.get(trip.attributes, "headsign")
+    end)
+    |> Enum.uniq()
   end
 
-  defp departure({:error, [%JsonApi.Error{code: "no_service"}]}) do
+  defp get_headsigns(%JsonApi{data: data_0}, %JsonApi{data: data_1}, :rapid_transit) do
+    headsign_0 = get_terminal_stops(data_0)
+    headsign_1 = get_terminal_stops(data_1)
+
+    {:ok, Enum.concat(headsign_0, headsign_1)}
+  end
+
+  defp get_headsigns(%JsonApi{data: _data_0}, %JsonApi{data: _data_1}, _) do
+    {:ok, []}
+  end
+
+  defp get_headsigns({:error, _} = error, _, _) do
+    error
+  end
+
+  defp is_terminus?(stop_name, headsigns) do
+    Enum.member?(headsigns, stop_name)
+  end
+
+  defp departure(%JsonApi{data: data}, headsigns, description) do
+    {:ok, departure(data, headsigns, description)}
+  end
+
+  defp departure({:error, [%JsonApi.Error{code: "no_service"}]}, _, _) do
     {:ok, :no_service}
   end
 
-  defp departure({:error, _} = error) do
+  defp departure({:error, _} = error, _, _) do
     error
   end
 
-  defp departure([]) do
+  defp departure([], _, _) do
     :no_service
   end
 
-  defp departure(data) do
+  # This one returns an array of hours
+  defp departure(data, headsigns, :rapid_transit) do
+    # This data is already going only one direction so we don't need to worry about that
+    times_by_stop =
+      Enum.group_by(data, fn x ->
+        stop_array = Map.get(x.relationships, "stop")
+        # should be exactly one element
+        List.first(stop_array).id
+      end)
+
+    Enum.map(times_by_stop, fn {id, x} ->
+      stop = Stops.Repo.get!(id)
+
+      {min, max} =
+        x
+        |> Stream.map(&Timex.parse!(time(&1.attributes), "{ISO:Extended}"))
+        |> Enum.min_max_by(&DateTime.to_unix(&1, :nanosecond))
+
+      %Departures{
+        stop_id: id,
+        first_departure: min,
+        last_departure: max,
+        stop_name: stop.name,
+        is_terminus: is_terminus?(stop.name, headsigns)
+      }
+    end)
+  end
+
+  # This returns a single hours map
+  defp departure(data, _headsigns, _description) do
     {min, max} =
       data
       |> Stream.map(&Timex.parse!(time(&1.attributes), "{ISO:Extended}"))
       |> Enum.min_max_by(&DateTime.to_unix(&1, :nanosecond))
 
     %Departures{
       first_departure: min,
-      last_departure: max
+      last_departure: max,
+      stop_name: nil,
+      stop_id: nil,
+      is_terminus: nil
```

Comment:
**thought (non-blocking):** Technically you only need `%Departures{first_departure: min, last_departure: max}` here. Omitting `stop_name`, `stop_id`, and `is_terminus` here will produce the same result, because you added these keys to the `defstruct` in `Schedules.Departures` and their default value is `nil`.

Reply:
Thanks for the info.  Still learning some of the details of Elixir.